### PR TITLE
Add go binding project reference

### PR DESCRIPTION
**Pull request by songtianyi**
1. Fix typos in include/mxnet/c_predict_api.h
2. Add [go-mxnet-predictor](https://github.com/songtianyi/go-mxnet-predictor) project reference to READMEs

go-mxnet-predictor features:
* more api bindings
* more comments and details in api
* dependency less, no need Amalgamation
* offering docker file to build mxnet&go dev env. So it's easy to build and run inference examples.

I am working at cloud service provider company. Our tech stack is based on go. It's painful to do machine learning in golang with mxnet, even with tensorflow. I am trying to fill the gap.
The prior project [gomxnet](https://github.com/jdeng/gomxnet) is outdated without maintaining.
PLZ review.

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ If you want to contribute to this list and the examples, please open a new pull
 * [MXNet Python](http://mxnet.readthedocs.io/en/latest/api/python/index.html) - Python library
 * [MXNetR](http://mxnet.readthedocs.io/en/latest/api/r/index.html) - R library
 * [MXNet.jl](http://mxnetjl.readthedocs.org/en/latest/) - Julia library
-* [gomxnet](https://github.com/jdeng/gomxnet) - Go binding
```
Comment (piiswrong) [inline example/README.md]: add it [Deprecated] or [Outdated] at the end instead of removing it

**Comment by piiswrong** [comment]
Thanks for the contribution. Please add a [Deprecated] or [Outdated] marker at the end instead of removing it. People may still want to use it as a reference.

If you are up to maintaining it for an extended period, consider merging it into main repo and add test cases to ensure it won't break.

**Comment by piiswrong** [comment]
The upcoming 0.9 release has some api changes. Please check if this is affected

File: example/README.md (hunk 1 above)
Comment (songtianyi) [inline example/README.md]: Ok got it

**Comment by songtianyi** [comment]
@piiswrong 
1. As you know **go-mxnet-predictor** is not a complete go-package like R and python.
Developers may feel confused about it, so it's not good to merge this into main repo currently in my opinion,  although I am willing to do that :)
2. 0.9rc seems not ready? NNVM was not tracked automatically in .git/config and make command didn't work well.
I will check **go-mxnet-predictor** immediately when 0.9 released.

**Comment by piiswrong** [comment]
you can put it together with the cpp prediction interface

**Comment by songtianyi** [comment]
It seems go-mxnet-predictor can't work with MXNet 0.9.
Main error message
```
panic: Failed loading Op convolution0 of type Convolution: [08:52:47] src/core/op.cc:55: Check failed: op != nullptr Operator Convolution is not registered
```

I saw the main change is the api code must be compiled by c++ now?
````
*! \brief Inhibit C++ name-mangling for MXNet functions. */
#ifdef __cplusplus
extern "C" {
#endif  // __cplusplus
````

**Comment by piiswrong** [comment]
see https://github.com/dmlc/mxnet/issues/4522

**Comment by songtianyi** [comment]
@piiswrong 
check pass, no panic any more.

**Pull request merged**